spec/jobs/subscription_confirm_job_spec.rb

```diff
@@ -134,6 +134,48 @@
       expect(job).to receive(:record_order)
     end
 
+    context "when Stripe payments need to be processed" do
+      let(:charge_response_mock) do
+        { status: 200, body: JSON.generate(id: "ch_1234", object: "charge", amount: 2000) }
+      end
+
+      before do
+        allow(order).to receive(:payment_required?) { true }
+        expect(job).to receive(:setup_payment!) { true }
+        stub_request(:post, "https://api.stripe.com/v1/charges")
+          .with(body: /amount/)
+          .to_return(charge_response_mock)
+      end
+
+      context "Stripe SCA" do
+        let(:stripe_sca_payment_method) { create(:stripe_sca_payment_method) }
+        let(:stripe_sca_payment) { create(:payment, amount: 10, payment_method: stripe_sca_payment_method) }
+
+        before do
+          allow(order).to receive(:pending_payments) { [stripe_sca_payment] }
+          expect(stripe_sca_payment_method).to receive(:charge_offline)
```

Comment:
this is the actual assertion of the test so it should rather be within the `it` block IMO. https://thoughtbot.com/blog/four-phase-test is a good article on this topic.

Reply:
That was a lot of head scratching for what turned out to be a change in tense 😂 - e749f75b10cd79cfc13780eba8b2182a9d1021d0